# Analyzing New York City's Evictions for a Non-profit Funding Program

>## Objective:
Find the most affected areas in New York City by evictions in order to allocate percentages of a 2 million dollar grant appropriately.

## Summary
**To help guide my analysis for self-practice, I created a scenario that will help to create some guiding questions.**

A non-profit from New York City received a grant for 2 million dollars to help alleviate the eviction crises occurring in NYC and prevent future displaced families after the eviction moratorium was lifted. I was tasked with composing a list of recommendations for the non-profit to aid them in dispersing the funds from the grant accordingly.  I needed to figure out where to allocate a larger percentages of the grant by finding out which Boroughs and Neighborhood Tabulation Areas were affected the most. Provided with data from NYC evictions between 2017 - 2022 from [NYC OpenData](https://opendata.cityofnewyork.us/), I explored and analyzed the dataset and came up with some recommendations on how to allocate the grant to assist in preventing future evictions in the new year.

**Some questions I asked the client before digging into the data:**
- What exactly do you want to find out?
- Where did your data come from?
- Is the data accurate?
                
**Some Questions I came accross when exploring the data and served as guiding questions to  my conclusions were:**

* Which borough had the highest eviction rate for the past 5 years?
* Which borough had consistently high eviction rates throughout the past 5 years?
* Which Neighborhood Tabulation Areas in those boroughs had the highest rate of evictions?
* Which part of the year are evictions high?
* Are there specific months that have high eviction rates?
* What are some recommendations on how to allocate the funding from the grant? 

**My Conclusion and recommendation:**


 ## Table of Contents  
1. [Open the data file and study the general information](#1)    
2. [Data Processing](#2)    
    2.1 [Replace the spaces in the column names](#3)<br>
    2.2 [Rename some column names](#4)<br>
    2.3 [Find Duplicates](#5)<br>
    2.4 [Drop Duplicates](#6)<br>
    2.5 [Convert Data Types](#7)<br>
    2.6 [Find Missing values](#8)<br>
    2.7 [Remove Columns](#9)
3. [Create new CSV](#10) 
4. [Guiding Questions](#11)     
    4.1 [Which borough had the highest eviction rate for the past 5 years?](#12)<br>
    4.2 [Which borough had consistently high eviction rates throughout the past 5 years?](#13)<br>
    4.3 [Which Neighborhood Tabulation Areas in those boroughs had the highest rate of evictions?](#14)<br>
    4.4 [Which part of the year are evictions high?](#15)<br>
5. [Overall Conclusion](#16)
6. [Recommendations](#17)


### Open the data file and study the general information <a class="anchor" id="1"></a>

In [1]:
import pandas as pd

pd.options.mode.chained_assignment = None  # default='warn'

print('Project libraries have been successfully imported!')


Project libraries have been successfully imported!


In [5]:
df = pd.read_csv('../csv-files/evictionsData.csv')
df

Court Index Number  Docket Number       Eviction Address  \
0              64723/17B           67848  75-04 JAMAICA AVENUE   
1              B10370/17           99907        3480 THIRD AVE   
2               77402/18           22745   112-10 111TH AVENUE   
3              50987/17B          485635      1038 BOSTON ROAD   
4               82534/16           66285  238 EAST 26TH STREET   
...                  ...             ...                   ...   
71921           69888/19           26372    93-01 197TH STREET   
71922           60297/19           24239         39-21 65TH PL   
71923           52350/18           81109  321 EAST 22ND STREET   
71924         R 52449/17           69331   98 TAYLOR STREET #6   
71925           48560/17          474931     1171 ELDER AVENUE   

      Eviction Apartment Number Executed Date Marshal First Name  \
0                            2F       1/29/18             Justin   
1                            9D       10/2/17            Darlene   
2                           NaN       5/14/19             Edward   
3                           4-C      12/18/18              Danny   
4                            C5        1/4/17              Henry   
...                         ...           ...                ...   
71921                       NaN        2/4/20             Edward   
71922                        GF      10/24/19             Edward   
71923                        5J        4/5/18              Henry   
71924                       NaN      10/18/17             Steven   
71925                        E5      11/29/17              Danny   

      Marshal Last Name Residential/Commercial        BOROUGH  \
0              Grossman            Residential         QUEENS   
1                Barone            Residential          BRONX   
2                 Guida            Residential         QUEENS   
3              Weinheim            Residential          BRONX   
4                 Daley            Residential       BROOKLYN   
...                 ...                    ...            ...   
71921             Guida            Residential         QUEENS   
71922             Guida            Residential         QUEENS   
71923             Daley            Residential      MANHATTAN   
71924            Powell            Residential  STATEN ISLAND   
71925          Weinheim            Residential          BRONX   

       Eviction Postcode         Ejectment Eviction/Legal Possession  \
0                  11421  Not an Ejectment                Possession   
1                  10456  Not an Ejectment                Possession   
2                  11420  Not an Ejectment                Possession   
3                  10456  Not an Ejectment                Possession   
4                  11226  Not an Ejectment                Possession   
...                  ...               ...                       ...   
71921              11423  Not an Ejectment                Possession   
71922              11377  Not an Ejectment                Possession   
71923              10012  Not an Ejectment                Possession   
71924              10310  Not an Ejectment                Possession   
71925              10472  Not an Ejectment                Possession   

        Latitude  Longitude  Community Board  Council District  Census Tract  \
0      40.691440 -73.866348              9.0              30.0           4.0   
1      40.830169 -73.906233              3.0              16.0         145.0   
2      40.678886 -73.826995             10.0              28.0         100.0   
3      40.826269 -73.906304              3.0              16.0         185.0   
4      40.643933 -73.952730             17.0              40.0         792.0   
...          ...        ...              ...               ...           ...   
71921  40.712416 -73.762604             12.0              27.0         500.0   
71922  40.745405 -73.898900              2.0              26.0         263.0   
71923  40.736934 -73.980684   

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 71926 entries, 0 to 71925
Data columns (total 20 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Court Index Number         71926 non-null  object 
 1   Docket Number              71926 non-null  int64  
 2   Eviction Address           71926 non-null  object 
 3   Eviction Apartment Number  59913 non-null  object 
 4   Executed Date              71926 non-null  object 
 5   Marshal First Name         71926 non-null  object 
 6   Marshal Last Name          71926 non-null  object 
 7   Residential/Commercial     71926 non-null  object 
 8   BOROUGH                    71926 non-null  object 
 9   Eviction Postcode          71926 non-null  int64  
 10  Ejectment                  71926 non-null  object 
 11  Eviction/Legal Possession  71926 non-null  object 
 12  Latitude                   65015 non-null  float64
 13  Longitude                  65015 non-null  flo

In [7]:
df.shape

(71926, 20)

### Conclusion


## Data Processing <a class="anchor" id="2"></a>

### Replace the spaces in the column names  <a class="anchor" id="3"></a>
This will make it easier to use when analzing the data

In [8]:
#replace the spaces between the column names with "_" using rename function
df.columns = df.columns.str.replace(' ','_')
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 71926 entries, 0 to 71925
Data columns (total 20 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Court_Index_Number         71926 non-null  object 
 1   Docket_Number_             71926 non-null  int64  
 2   Eviction_Address           71926 non-null  object 
 3   Eviction_Apartment_Number  59913 non-null  object 
 4   Executed_Date              71926 non-null  object 
 5   Marshal_First_Name         71926 non-null  object 
 6   Marshal_Last_Name          71926 non-null  object 
 7   Residential/Commercial     71926 non-null  object 
 8   BOROUGH                    71926 non-null  object 
 9   Eviction_Postcode          71926 non-null  int64  
 10  Ejectment                  71926 non-null  object 
 11  Eviction/Legal_Possession  71926 non-null  object 
 12  Latitude                   65015 non-null  float64
 13  Longitude                  65015 non-null  flo

### Rename some column names  <a class="anchor" id="4"></a>
For better uniformity

In [9]:
#renemaing the selected columns with rename function
df = df.rename(
    columns = {'Docket_Number_':'Docket_Number',
               'Residential/Commercial':'Property_Type',
               'BOROUGH':'Borough',
               'NTA': 'Neighborhood_Tabulation_Area'})
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 71926 entries, 0 to 71925
Data columns (total 20 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Court_Index_Number            71926 non-null  object 
 1   Docket_Number                 71926 non-null  int64  
 2   Eviction_Address              71926 non-null  object 
 3   Eviction_Apartment_Number     59913 non-null  object 
 4   Executed_Date                 71926 non-null  object 
 5   Marshal_First_Name            71926 non-null  object 
 6   Marshal_Last_Name             71926 non-null  object 
 7   Property_Type                 71926 non-null  object 
 8   Borough                       71926 non-null  object 
 9   Eviction_Postcode             71926 non-null  int64  
 10  Ejectment                     71926 non-null  object 
 11  Eviction/Legal_Possession     71926 non-null  object 
 12  Latitude                      65015 non-null  float64
 13  L

### Find Duplicates  <a class="anchor" id="5"></a>

In [10]:
#Count how many rows are duplicates
df.duplicated().value_counts().to_frame('count')

count
False  71733
True     193

In [11]:
#Show  a list of some duplicates
df[df.duplicated()]

Court_Index_Number  Docket_Number  \
504             80748/18          13692   
1586            64645/19          27455   
1942           304104/21         360808   
4990            26181/19         355347   
5250            50942/20          27898   
...                  ...            ...   
69628           31962/18         348511   
70655           22633/19         355950   
71444              95038          17860   
71711           22645/19         355486   
71822           18503/19         355267   

                                 Eviction_Address Eviction_Apartment_Number  \
504    79-81 WHITE STREET - #1W (ENTIRE PREMISES)                       NaN   
1586                                23-44 31ST RD                        2R   
1942                            260 SPRING STREET                   STORAGE   
4990                     2160 EAST TREMONT AVENUE                        3B   
5250                       98-48 QUEENS BOULEVARD                       NaN   
...                                           ...                       ...   
69628                        2145 STARLING AVENUE                  UNIT 629   
70655                       825 EAST 217TH STREET                   ROOM#1,   
71444                         1903 - OCEAN AVENUE                        D1   
71711                           1025 MANOR AVENUE                        2F   
71822                          3652 HARPER AVENUE                  BASEMENT   

      Executed_Date Marshal_First_Name Marshal_Last_Name Property_Type  \
504         8/18/21             Robert          Renzulli    Commercial   
1586        1/26/22             Edward             Guida   Residential   
1942        12/6/21             Thomas               Bia    Commercial   
4990        12/3/19             Thomas               Bia   Residential   
5250        10/8/21             Edward             Guida    Commercial   
...             ...                ...               ...           ...   
69628        3/5/20             Thomas               Bia   Residential   
70655       11/4/19             Thomas               Bia   Residential   
71444       2/16/17            Charles       Marchisotto   Residential   
71711      12/13/19             Thomas               Bia   Residential   
71822        1/8/20             Thomas               Bia   Residential   

         Borough  Eviction_Postcode         Ejectment  \
504    MANHATTAN              10013  Not an Ejectment   
1586      QUEENS              11106  Not an Ejectment   
1942   MANHATTAN              10013  Not an Ejectment   
4990       BRONX              10462  Not an Ejectment   
5250      QUEENS              11374  Not an Ejectment   
...          ...                ...               ...   
69628      BRONX              10462  Not an Ejectment   
70655      BRONX              10467  Not an Ejectment   
71444   BROOKLYN              11230  Not an Ejectment   
71711      BRONX              10472  Not an Ejectment   
71822      BRONX              10466  Not an Ejectment   

      Eviction/Legal_Possession   Latitude  Longitude  Community_Board  \
504                  Possession        NaN        NaN              NaN   
1586                 Possession  40.765769 -73.928785              1.0   
1942                 Possession  40.725704 -74.006195              2.0   
4990                 Possession  40.842096 -73.856561              9.0   
5250                 Possession  40.727948 -73.856374              6.0   
...                         ...        ...        ...              ...   
69628                Possession  40.836210 -73.853708              9.0   
70655                Possession  40.880962 -73.859776             12.0   
71444                Possession        NaN        NaN              NaN   
71711                Possession  40.824911 -73.875459              9.0   
71822                Possession  40.885920 -73.833440             12.0   

       Council_District  Census_Tract        BIN           BBL  \
504        

**I wanted to double check that the duplicates found were indeed duplicates and where they were indexed at**

In [12]:
df.loc[(df['Court_Index_Number']== '80748/18')]

Court_Index_Number  Docket_Number  \
503             80748/18          13692   
504             80748/18          13692   
30420           80748/18          11964   

                                 Eviction_Address Eviction_Apartment_Number  \
503    79-81 WHITE STREET - #1W (ENTIRE PREMISES)                       NaN   
504    79-81 WHITE STREET - #1W (ENTIRE PREMISES)                       NaN   
30420  79-81 WHITE STREET - #1W (ENTIRE PREMISES)                       NaN   

      Executed_Date Marshal_First_Name Marshal_Last_Name Property_Type  \
503         8/18/21             Robert          Renzulli    Commercial   
504         8/18/21             Robert          Renzulli    Commercial   
30420      10/31/19             Robert          Renzulli    Commercial   

         Borough  Eviction_Postcode         Ejectment  \
503    MANHATTAN              10013  Not an Ejectment   
504    MANHATTAN              10013  Not an Ejectment   
30420  MANHATTAN              10013  Not an Ejectment   

      Eviction/Legal_Possession  Latitude  Longitude  Community_Board  \
503                  Possession       NaN        NaN              NaN   
504                  Possession       NaN        NaN              NaN   
30420                Possession       NaN        NaN              NaN   

       Council_District  Census_Tract  BIN  BBL Neighborhood_Tabulation_Area  
503                 NaN           NaN  NaN  NaN                          NaN  
504                 NaN           NaN  NaN  NaN                          NaN  
30420               NaN           NaN  NaN  NaN                          NaN

**Not all rows were duplicates as you can see above. Index 30420 shows that the eviction for this was executed in October of 2019, instead of 8/18/21.**
So, for this reason I decided to rerun the duplicated function, but this time I want to look at specific columns that will give a clearer picture on whether or not these are indeed duplicates. 

In [13]:
df[df.duplicated(
    ['Court_Index_Number', 
     'Executed_Date', 
     'Eviction_Address', 
     'Marshal_Last_Name', ])]


Court_Index_Number  Docket_Number  \
504             80748/18          13692   
1257            75256/17          16181   
1586            64645/19          27455   
1590            61523/18          21324   
1653            19898/18         349008   
...                  ...            ...   
71845           89880/16           8415   
71859           68668/17           6474   
71884           15185/18          18683   
71909           63248/18           8549   
71921           69888/19          26372   

                                 Eviction_Address Eviction_Apartment_Number  \
504    79-81 WHITE STREET - #1W (ENTIRE PREMISES)                       NaN   
1257                        221-23 MURDOCK AVENUE                       NaN   
1586                                23-44 31ST RD                        2R   
1590                             109-48 126TH ST.                       NaN   
1653                             1660 BOSTON ROAD                       6-C   
...                                           ...                       ...   
71845                         2319 BEDFORD AVENUE                       NaN   
71859                            1079 HANCOCK ST.                   2ND FL.   
71884                        1700 GRAND CONCOURSE                       12E   
71909                         137-07 253RD STREET                       NaN   
71921                          93-01 197TH STREET                       NaN   

      Executed_Date Marshal_First_Name Marshal_Last_Name Property_Type  \
504         8/18/21             Robert          Renzulli    Commercial   
1257         5/7/18             Edward             Guida   Residential   
1586        1/26/22             Edward             Guida   Residential   
1590        2/11/19             Edward             Guida   Residential   
1653        3/20/19             Thomas               Bia   Residential   
...             ...                ...               ...           ...   
71845       3/16/17             Edward             Guida   Residential   
71859       9/27/17             Robert          Renzulli   Residential   
71884       3/12/19             Edward             Guida   Residential   
71909      11/21/18            Bernard             Blake   Residential   
71921        2/4/20             Edward             Guida   Residential   

         Borough  Eviction_Postcode         Ejectment  \
504    MANHATTAN              10013  Not an Ejectment   
1257      QUEENS              11429  Not an Ejectment   
1586      QUEENS              11106  Not an Ejectment   
1590      QUEENS              11420  Not an Ejectment   
1653       BRONX              10460  Not an Ejectment   
...          ...                ...               ...   
71845   BROOKLYN              11226  Not an Ejectment   
71859   BROOKLYN              11221  Not an Ejectment   
71884      BRONX              10457  Not an Ejectment   
71909     QUEENS              11422  Not an Ejectment   
71921     QUEENS              11423  Not an Ejectment   

      Eviction/Legal_Possession   Latitude  Longitude  Community_Board  \
504                  Possession        NaN        NaN              NaN   
1257                 Possession  40.703263 -73.735399             13.0   
1586                 Possession  40.765769 -73.928785              1.0   
1590                 Possession  40.683702 -73.815892             10.0   
1653                 Possession  40.836039 -73.889395              3.0   
...                         ...        ...        ...              ...   
71845                Possession  40.647173 -73.955676             17.0   
71859                Possession  40.688942 -73.914415              4.0   
71884                Possession  40.844479 -73.911464              4.0   
71909                Possession  40.662407 -73.728306             13.0   
71921                Possession  40.712416 -73.762604             12.0   

       Council_District  Census_Tract        BIN           BBL  \
504        

Lets check one of these rows for their duplicates again.

In [14]:
df.loc[(df['Court_Index_Number']== '69888/19')]

Court_Index_Number  Docket_Number    Eviction_Address  \
5068            69888/19          26371  93-01 197TH STREET   
71921           69888/19          26372  93-01 197TH STREET   

      Eviction_Apartment_Number Executed_Date Marshal_First_Name  \
5068                        NaN        2/4/20             Edward   
71921                       NaN        2/4/20             Edward   

      Marshal_Last_Name Property_Type Borough  Eviction_Postcode  \
5068              Guida   Residential  QUEENS              11423   
71921             Guida   Residential  QUEENS              11423   

              Ejectment Eviction/Legal_Possession   Latitude  Longitude  \
5068   Not an Ejectment                Possession  40.712416 -73.762604   
71921  Not an Ejectment                Possession  40.712416 -73.762604   

       Community_Board  Council_District  Census_Tract        BIN  \
5068              12.0              27.0         500.0  4231565.0   
71921             12.0              27.0         500.0  4231565.0   

                BBL Neighborhood_Tabulation_Area  
5068   4.108270e+09                       Hollis  
71921  4.108270e+09                       Hollis

### Conclusion
In attempting to find all duplicates with the duplicated() function alone, I only retrieved 193 rows of duplicates. When double checking one of the rows for its duplicate, I found that it was possible for some of the rows to have a different *Executed_Date*, but the same *Court_Index_Number*. I decided to rerun the duplicated() function with specific columns like
* *Court_Index_Number*
* *Executed_Date*
* *Eviction_Address* 
* *Marshal_Last_Name*

and this this it returned over 2k rows of duplicates. Some of the duplicates have different *Docket_Number*s, but all other columns are the same. 

### Drop Duplicates  <a class="anchor" id="6"></a>

In [15]:
df = df.drop_duplicates(subset=[
    'Court_Index_Number', 
    'Executed_Date', 
    'Eviction_Address', 
    'Marshal_Last_Name', ], keep=False)


In [16]:
df.duplicated().value_counts().to_frame('count')

count
False  67659

In [17]:
df.shape

(67659, 20)

### Convert Data Types  <a class="anchor" id="7"></a>

- *Executed_Date* to a datatime data type.
- *Council_District* to an Int64
- *Census_Tract* to an Int64

I decided to convert *Council_District* and *Census_Tract* to Int64 instead of int64 because of the null values in those columns. 


In [18]:
#Convert to datetime type
df['Executed_Date'] = pd.to_datetime(df['Executed_Date'])
#Convert to int64
df['Community_Board'] = df['Community_Board'].astype(float).astype('Int64')
df['Council_District'] = df['Council_District'].astype(float).astype('Int64')
df['Census_Tract'] = df['Census_Tract'].astype(float).astype('Int64')
#Confirm change
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 67659 entries, 0 to 71925
Data columns (total 20 columns):
 #   Column                        Non-Null Count  Dtype         
---  ------                        --------------  -----         
 0   Court_Index_Number            67659 non-null  object        
 1   Docket_Number                 67659 non-null  int64         
 2   Eviction_Address              67659 non-null  object        
 3   Eviction_Apartment_Number     57215 non-null  object        
 4   Executed_Date                 67659 non-null  datetime64[ns]
 5   Marshal_First_Name            67659 non-null  object        
 6   Marshal_Last_Name             67659 non-null  object        
 7   Property_Type                 67659 non-null  object        
 8   Borough                       67659 non-null  object        
 9   Eviction_Postcode             67659 non-null  int64         
 10  Ejectment                     67659 non-null  object        
 11  Eviction/Legal_Possession   

### Find Missing values  <a class="anchor" id="8"></a>

In [19]:
#finding the total null values in each column
df.isna().sum()

Court_Index_Number                  0
Docket_Number                       0
Eviction_Address                    0
Eviction_Apartment_Number       10444
Executed_Date                       0
Marshal_First_Name                  0
Marshal_Last_Name                   0
Property_Type                       0
Borough                             0
Eviction_Postcode                   0
Ejectment                           0
Eviction/Legal_Possession           0
Latitude                         6584
Longitude                        6584
Community_Board                  6584
Council_District                 6584
Census_Tract                     6584
BIN                              6705
BBL                              6705
Neighborhood_Tabulation_Area     6584
dtype: int64

There are null values under several columns. I'm not too concerned with the null values located in the Eviction_Apartment_Number column because some of these are either residential homes or comercial propperties. For this reason I will be removing that column later in my analysis.

However, I am a litte interested in why there are 6,584 null values in columns: 
- Latitude
- Longitude
- Community_Board
- Council_District
- Census_Tract
- Neighborhood_Tabulation_Area

and 6,705 null values in columns:
- BIN
- BBL

**Let's Investigate!**

I want to see what these null values look like in the table, so I will look into null values found under the Latitude column first.

In [20]:
df[df['Latitude'].isna()]

Court_Index_Number  Docket_Number  \
11            R 50581/17          68539   
15              73903/15         151310   
34              73546/18          86004   
40              76144/16         292067   
49              66704/17          23030   
...                  ...            ...   
71887         B901492/19         102017   
71902           77244/17         154664   
71904           11024/17         152894   
71914           94125/17         154184   
71917           92962/17          80738   

                                        Eviction_Address  \
11                     457-461 PORT RICHMOND AVE.(MAIN E   
15                  193-14 NASHVILLE BLVD.(ENTIRE HOUSE)   
34                         2107-2109 EAST   18T H STREET   
40                    599 WEST 176TH STREET BEAUTY SALON   
49                       1733 E 55 STREET  APT 1ST FLOOR   
...                                                  ...   
71887  597 SOUTHERN BOULEVA RD B/K/A 585-605 SOUTHERN...   
71902                    104-17 177TH STREET (ROOM #3R )   
71904                     109-45 FARMERS BLVD. (STORE 2)   
71914                        490 NEW YORK AVE. (UNIT L9)   
71917                 2146 BEVERLEY ROAD ENTIRE PREMISES   

      Eviction_Apartment_Number Executed_Date Marshal_First_Name  \
11                          NaN    2017-05-02             Steven   
15                          NaN    2017-03-31             Ronald   
34             COMMERCIAL SPACE    2018-10-24             Justin   
40                         ST#9    2017-04-07             George   
49                          NaN    2017-09-26               Gary   
...                         ...           ...                ...   
71887                       NaN    2020-02-18             Ileana   
71902                       NaN    2018-07-13             Ronald   
71904                       NaN    2018-02-09             Ronald   
71914                       NaN    2018-05-09             Ronald   
71917                       NaN    2018-04-02              Henry   

      Marshal_Last_Name Property_Type        Borough  Eviction_Postcode  \
11               Powell    Commercial  STATEN ISLAND              10302   
15               Pazant   Residential         QUEENS              11413   
34             Grossman    Commercial       BROOKLYN              11229   
40               Essock    Commercial      MANHATTAN              10033   
49                 Rose   Residential       BROOKLYN              11234   
...                 ...           ...            ...                ...   
71887            Rivera    Commercial          BRONX              10455   
71902            Pazant   Residential         QUEENS              11434   
71904            Pazant   Residential         QUEENS              11412   
71914            Pazant   Residential       BROOKLYN              11225   
71917             Daley    Commercial       BROOKLYN              11226   

              Ejectment Eviction/Legal_Possession  Latitude  Longitude  \
11     Not an Ejectment                Possession       NaN        NaN   
15     Not an Ejectment                Possession       NaN        NaN   
34     Not an Ejectment                Possession       NaN        NaN   
40     Not an Ejectment                Possession       NaN        NaN   
49     Not an Ejectment                Possession       NaN        NaN   
...                 ...                       ...       ...        ...   
71887  Not an Ejectment                Possession       NaN        NaN   
71902  Not an Ejectment                Possession       NaN        NaN   
71904  Not an Ejectment                Possession       NaN        NaN   
71914  Not an Ejectment                Possession       NaN        NaN   
71917  Not an Ejectment                Possession       NaN        NaN   

       Community_Board  Council_District  Census_Tract  BIN  BBL  \
11                <NA>              <NA>          <NA>  NaN  NaN   
15                <NA>   

Now, I am curious about the BIN, BBL columns since they have 121 more null values. So let's take a look at the rows that only have a null value in both BIN, BBL columns ONLY.

In [21]:
#pd.set_option("display.max_rows", None)
df.loc[(df['BIN'].isna()) & (df['BBL'].isna()) & (df['Latitude'].notna())]

Court_Index_Number  Docket_Number         Eviction_Address  \
1897            67961/17          12419   103-18A NORTHERN BLVD.   
2365            65390/17          12536  179-05 LINDEN BOULEVARD   
2965           55921/18C           3933    173-29 VASWANI AVENUE   
3926           900486/19         357149   170B WEST 233RD STREET   
4232          K100843/16         377435          4 VERONA STREET   
...                  ...            ...                      ...   
70705           81412/16          68912      110-26A LIBERTY AVE   
71216           52639/17          76407        19B OLYMPIA BLVD.   
71338         52004/18-2         205555       147-31 230TH PLACE   
71346          900463/19          95073     1797 WESTCHESTER AVE   
71403           68498/18         206341        58-75 54TH AVENUE   

      Eviction_Apartment_Number Executed_Date Marshal_First_Name  \
1897                        NaN    2017-10-04             Edward   
2365                        NaN    2018-07-16             Edward   
2965                         2A    2018-09-13              Vadim   
3926                        NaN    2019-10-31             Thomas   
4232                        10B    2017-03-22            Richard   
...                         ...           ...                ...   
70705           1ST FLOOR STORE    2017-11-06              Henry   
71216                       NaN    2018-01-19            Kenneth   
71338                       NaN    2018-08-20            Richard   
71346              UNIT #102-12    2020-02-11              Henry   
71403                       NaN    2018-11-05            Richard   

      Marshal_Last_Name Property_Type        Borough  Eviction_Postcode  \
1897              Guida    Commercial         QUEENS              11368   
2365              Guida    Commercial         QUEENS              11434   
2965        Barbarovich   Residential         QUEENS              11434   
3926                Bia    Commercial          BRONX              10463   
4232              McCoy   Residential       BROOKLYN              11231   
...                 ...           ...            ...                ...   
70705             Daley    Commercial         QUEENS              11419   
71216         Giachetta    Commercial  STATEN ISLAND              10305   
71338           Capuano   Residential         QUEENS              11413   
71346             Daley    Commercial          BRONX              10472   
71403           Capuano    Commercial         QUEENS              11378   

              Ejectment Eviction/Legal_Possession   Latitude  Longitude  \
1897   Not an Ejectment                Possession  40.757640 -73.865377   
2365   Not an Ejectment                Possession  40.692429 -73.767875   
2965   Not an Ejectment                Possession  40.682282 -73.769746   
3926   Not an Ejectment                Possession  40.880298 -73.902738   
4232   Not an Ejectment                Possession  40.677709 -74.006435   
...                 ...                       ...        ...        ...   
70705  Not an Ejectment                Possession  40.684528 -73.831665   
71216  Not an Ejectment                Possession  40.592803 -74.069014   
71338  Not an Ejectment                Possession  40.657198 -73.751426   
71346  Not an Ejectment                Possession  40.831826 -73.866557   
71403  Not an Ejectment                Possession  40.729945 -73.909427   

       Community_Board  Council_District  Census_Tract  BIN  BBL  \
1897                 3                21           379  NaN  NaN   
2365                12                27           424  NaN  NaN   
2965                12                27         33401  NaN  NaN   
3926                 8                11           283  NaN  NaN   
4232                 6                38            59  NaN  NaN   
...                ...               ...           ...  ...  ...   
70705               10                28           100  NaN  NaN   
71216                2    

### Conclusion

My observations show that there are no Latitude or Longitude values provided when the following columns after them (*Community_Board, Council_District, Census_Tract, BIN, BBL, Neighborhood_Tabulation_Area*) are also null. This doesnt explain why columns BIN and BBL have an extra 121 rows with null values, however, after some outside research it seems that not all buildings have a BIN or BBL.

For these reasons, I don't see the need to remove these rows out of our analysis yet. These missing values will not effect my analysis for deciding how much of the grant to allocate to each borough because these rows all have a value under the *Borough* column.

### Remove Columns  <a class="anchor" id="9"></a>

I decided to remove the following columns:

- *Docket_Number* - Because the *Docket_Number* are catergorical values that doesn't serve a purpose for my analysis. I decided to keep the *Court_Index_Number* instead so it can serve as an extra index identifier. 

- *Eviction_Apartment_Number* - I removed the *Eviction_Apartment_Number* because whether the eviction was from an apartment or a residence, it has no effect on my recommendations for allocating the funds. 

- *Marshal_First_Name* and *Marshal_Last_Name* - I removed the marshals name out of my analysis as well since my focus is to find the most effected boroughs and which marshals executed the eviction makes no difference.

- *Census_Tract* - this is roughly the size of a neighborhood, but may not correspond to actual neighborhoods or city boundaries. I will focus on the *Neighborhood_Tabulation_Area* instead.

- *BIN* and *BBL* - These also served no use in my analysis.

In [22]:
#lets use the columns function to make it easier to delete the ones I dont need
df.columns

Index(['Court_Index_Number', 'Docket_Number', 'Eviction_Address',
       'Eviction_Apartment_Number', 'Executed_Date', 'Marshal_First_Name',
       'Marshal_Last_Name', 'Property_Type', 'Borough', 'Eviction_Postcode',
       'Ejectment', 'Eviction/Legal_Possession', 'Latitude', 'Longitude',
       'Community_Board', 'Council_District', 'Census_Tract', 'BIN', 'BBL',
       'Neighborhood_Tabulation_Area'],
      dtype='object')

In [23]:
df_subset= df[['Court_Index_Number', 'Eviction_Address','Executed_Date', 'Property_Type', 
               'Borough', 'Eviction_Postcode','Ejectment', 'Eviction/Legal_Possession', 'Latitude', 
               'Longitude','Community_Board', 'Council_District', 'Neighborhood_Tabulation_Area']]
df_subset

Court_Index_Number           Eviction_Address Executed_Date  \
0              64723/17B       75-04 JAMAICA AVENUE    2018-01-29   
1              B10370/17             3480 THIRD AVE    2017-10-02   
2               77402/18        112-10 111TH AVENUE    2019-05-14   
3              50987/17B           1038 BOSTON ROAD    2018-12-18   
4               82534/16       238 EAST 26TH STREET    2017-01-04   
...                  ...                        ...           ...   
71920           54357/18  1223 HANCOCK STREET APT 3    2018-08-24   
71922           60297/19              39-21 65TH PL    2019-10-24   
71923           52350/18       321 EAST 22ND STREET    2018-04-05   
71924         R 52449/17        98 TAYLOR STREET #6    2017-10-18   
71925           48560/17          1171 ELDER AVENUE    2017-11-29   

      Property_Type        Borough  Eviction_Postcode         Ejectment  \
0       Residential         QUEENS              11421  Not an Ejectment   
1       Residential          BRONX              10456  Not an Ejectment   
2       Residential         QUEENS              11420  Not an Ejectment   
3       Residential          BRONX              10456  Not an Ejectment   
4       Residential       BROOKLYN              11226  Not an Ejectment   
...             ...            ...                ...               ...   
71920   Residential       BROOKLYN              11221  Not an Ejectment   
71922   Residential         QUEENS              11377  Not an Ejectment   
71923   Residential      MANHATTAN              10012  Not an Ejectment   
71924   Residential  STATEN ISLAND              10310  Not an Ejectment   
71925   Residential          BRONX              10472  Not an Ejectment   

      Eviction/Legal_Possession   Latitude  Longitude  Community_Board  \
0                    Possession  40.691440 -73.866348                9   
1                    Possession  40.830169 -73.906233                3   
2                    Possession  40.678886 -73.826995               10   
3                    Possession  40.826269 -73.906304                3   
4                    Possession  40.643933 -73.952730               17   
...                         ...        ...        ...              ...   
71920                Possession  40.691934 -73.911483                4   
71922                Possession  40.745405 -73.898900                2   
71923                Possession  40.736934 -73.980684                6   
71924                Possession  40.636526 -74.123404                1   
71925                Possession  40.827687 -73.878974                9   

       Council_District               Neighborhood_Tabulation_Area  
0                    30                                  Woodhaven  
1                    16                         Claremont-Bathgate  
2                    28                           South Ozone Park  
3                    16                         Morrisania-Melrose  
4                    40                                    Erasmus  
...                 ...                                        ...  
71920                37                             Bushwick South  
71922                26                                   Woodside  
71923                 2                                   Gramercy  
71924                49  West New Brighton-New Brighton-St. George  
71925                17                         Soundview-Bruckner  

[67659 rows x 13 columns]

## Create new CSV <a class="anchor" id="10"></a>

Let's save our cleaned csv file we have been working on in it's own csv file for future use for our visualizations.

In [24]:
df_subset.to_csv('/Users/Mo/Desktop/NYC-Evictions-Analysis/csv-files/transformed_evictions_df.csv', index=False, header=True)


## Guiding Questions <a class="anchor" id="11"></a>

* Which borough had the highest eviction rate for the past 6 years?
* Which borough had consistently high eviction rates throughout the past 5 years?
* Which Neighborhood Tabulation Areas in those boroughs had the highest rate of evictions?
* Which part of the year are evictions high?
* Are there specific months that have high eviction rates?
* Whart are some recommendations on how to allocate the funding from the grant? 

**Question** - Which borough had the highest eviction rate for the past 5 years? <a class="anchor" id="12"></a>

In [25]:
## Find the highest eviction rate in each borough for the past 5 years
df_subset['Borough'].value_counts().to_frame()

Borough
BRONX            21963
BROOKLYN         20198
QUEENS           12216
MANHATTAN        10833
STATEN ISLAND     2449

#### Conclusion
The Bronx had the highest amount of evictions for the past 5 years with over 21k evictions.

**Question** - Which boroughs had consistently high eviction rates throughout the past 5 years? <a class="anchor" id="13"></a>

In [26]:
# creating a new column for year from executed date to use for next question
df_subset['Year'] = df_subset['Executed_Date'].dt.year

In [37]:
# removing rows from 2022 since there is not a full year of data to give an accurate comparable analysis
no_2023 = df_subset.query('Year != 2023')
# getting the eviction count for each borough per year
borough_year_count = no_2023.groupby('Year').Borough.value_counts().reset_index(name='Eviction_count')
borough_year_count

Year        Borough  Eviction_count
0   2017          BRONX            7186
1   2017       BROOKLYN            6183
2   2017         QUEENS            3739
3   2017      MANHATTAN            3368
4   2017  STATEN ISLAND             732
5   2018          BRONX            6781
6   2018       BROOKLYN            5983
7   2018         QUEENS            3878
8   2018      MANHATTAN            3276
9   2018  STATEN ISLAND             691
10  2019          BRONX            5858
11  2019       BROOKLYN            5139
12  2019         QUEENS            3211
13  2019      MANHATTAN            2725
14  2019  STATEN ISLAND             620
15  2020          BRONX             986
16  2020       BROOKLYN             960
17  2020         QUEENS             593
18  2020      MANHATTAN             496
19  2020  STATEN ISLAND             108
20  2021       BROOKLYN              82
21  2021      MANHATTAN              28
22  2021  STATEN ISLAND              27
23  2021         QUEENS              22
24  2021          BRONX              17
25  2022       BROOKLYN            1814
26  2022          BRONX            1113
27  2022      MANHATTAN             920
28  2022         QUEENS             763
29  2022  STATEN ISLAND             269

In [28]:
borough_year_count['%'] = borough_year_count['Eviction_count'].div(borough_year_count.groupby('Year')['Eviction_count'].transform('sum')).mul(100)
borough_year_count.sort_values(by=['Year','%'], ascending=False)

Year        Borough  Eviction_count          %
25  2022       BROOKLYN            1814  37.179750
26  2022          BRONX            1113  22.812052
27  2022      MANHATTAN             920  18.856323
28  2022         QUEENS             763  15.638451
29  2022  STATEN ISLAND             269   5.513425
20  2021       BROOKLYN              82  46.590909
21  2021      MANHATTAN              28  15.909091
22  2021  STATEN ISLAND              27  15.340909
23  2021         QUEENS              22  12.500000
24  2021          BRONX              17   9.659091
15  2020          BRONX             986  31.371301
16  2020       BROOKLYN             960  30.544066
17  2020         QUEENS             593  18.867324
18  2020      MANHATTAN             496  15.781101
19  2020  STATEN ISLAND             108   3.436207
10  2019          BRONX            5858  33.373213
11  2019       BROOKLYN            5139  29.277047
12  2019         QUEENS            3211  18.293169
13  2019      MANHATTAN            2725  15.524412
14  2019  STATEN ISLAND             620   3.532160
5   2018          BRONX            6781  32.903101
6   2018       BROOKLYN            5983  29.031006
7   2018         QUEENS            3878  18.817022
8   2018      MANHATTAN            3276  15.895968
9   2018  STATEN ISLAND             691   3.352904
0   2017          BRONX            7186  33.883440
1   2017       BROOKLYN            6183  29.154093
2   2017         QUEENS            3739  17.630140
3   2017      MANHATTAN            3368  15.880800
4   2017  STATEN ISLAND             732   3.451528

**Conclusion** 


Between 2017 - 2020 the Bronx Borough consistently had the highest eviction rates. In 2021 the Bronx had the lowest eviction rate. This could have been due to the Covid-19 pandemic that hit the world in 2020 and an eviction moratorium(Mar 7, 2020 - Jan 15, 2022) took effect, preventing evictions without rare circumstances.


**Question** - Which Neighborhood Tabulation Areas in those boroughs had the highest rate of evictions <a class="anchor" id="14"></a>

In [29]:
#counting how many unique nighborhoods we are working with
df_subset.Neighborhood_Tabulation_Area.nunique()

191

In [30]:
#removing retriction in order to few all rows
pd.set_option('display.max_rows', None)
#
top_NTA = no_2023.groupby(['Year','Borough']).Neighborhood_Tabulation_Area.value_counts().reset_index(name='Eviction_count')
top_NTA

Year        Borough                       Neighborhood_Tabulation_Area  \
0     2017          BRONX                         Bedford Park-Fordham North   
1     2017          BRONX                           Williamsbridge-Olinville   
2     2017          BRONX                                         Mount Hope   
3     2017          BRONX                  University Heights-Morris Heights   
4     2017          BRONX                   East Concourse-Concourse Village   
5     2017          BRONX                                 Morrisania-Melrose   
6     2017          BRONX                                       East Tremont   
7     2017          BRONX                                            Belmont   
8     2017          BRONX                                            Norwood   
9     2017          BRONX                              Van Cortlandt Village   
10    2017          BRONX                                      Fordham South   
11    2017          BRONX                                     West Concourse   
12    2017          BRONX                             Mott Haven-Port Morris   
13    2017          BRONX                                Kingsbridge Heights   
14    2017          BRONX                     Melrose South-Mott Haven North   
15    2017          BRONX                                         Highbridge   
16    2017          BRONX                             West Farms-Bronx River   
17    2017          BRONX                                 Claremont-Bathgate   
18    2017          BRONX                                          Bronxdale   
19    2017          BRONX                                  Crotona Park East   
20    2017          BRONX                                           Longwood   
21    2017          BRONX                                 Soundview-Bruckner   
22    2017          BRONX                                 Woodlawn-Wakefield   
23    2017          BRONX                                        Hunts Point   
24    2017          BRONX    Soundview-Castle Hill-Clason Point-Harding Park   
25    2017          BRONX                                        Parkchester   
26    2017          BRONX                    Eastchester-Edenwald-Baychester   
27    2017          BRONX                              Westchester-Unionport   
28    2017          BRONX                                     Pelham Parkway   
29    2017          BRONX            Van Nest-Morris Park-Westchester Square   
30    2017          BRONX                                         Co-op City   
31    2017          BRONX                         Spuyten Duyvil-Kingsbridge   
32    2017          BRONX           Schuylerville-Throgs Neck-Edgewater Park   
33    2017          BRONX                Pelham Bay-Country Club-City Island   
34    2017          BRONX                North Riverdale-Fieldston-Riverdale   
35    2017          BRONX                            Allerton-Pelham Gardens   
36    2017          BRONX                            park-cemetery-etc-Bronx   
37    2017          BRONX                                 Marble Hill-Inwood   
38    2017       BROOKLYN                                Crown Heights North   
39    2017       BROOKLYN                                      East New York   
40    2017       BROOKLYN                                           Flatbush   
41    2017       BROOKLYN                  Prospect Lefferts Gardens-Wingate   
42    2017       BROOKLYN                                        Brownsville   
43    2017       BROOKLYN                               Rugby-Remsen Village   
44    2017       BROOKLYN                                 Stuyvesant Heights   
45    2017       BROOKLYN                             East Flatbush-Farragut   
46    2017       BROOKLYN                                         Ocean Hill   
47    2017       BROOKLYN                                            Erasmus   
48    2017       BROOKLYN                   East New York (Pennsylvania Ave)   
49    

In [41]:
#removing retriction in order to few all rows
pd.set_option('display.max_rows', None)
#
top_NTA = no_2023.groupby('Borough').Neighborhood_Tabulation_Area.value_counts().reset_index(name='Eviction_count')
top_NTA

Borough                       Neighborhood_Tabulation_Area  \
0            BRONX                   East Concourse-Concourse Village   
1            BRONX                           Williamsbridge-Olinville   
2            BRONX                         Bedford Park-Fordham North   
3            BRONX                  University Heights-Morris Heights   
4            BRONX                                         Mount Hope   
5            BRONX                                 Morrisania-Melrose   
6            BRONX                                       East Tremont   
7            BRONX                             Mott Haven-Port Morris   
8            BRONX                                     West Concourse   
9            BRONX                                            Norwood   
10           BRONX                              Van Cortlandt Village   
11           BRONX                                            Belmont   
12           BRONX                             West Farms-Bronx River   
13           BRONX                     Melrose South-Mott Haven North   
14           BRONX                                      Fordham South   
15           BRONX                                          Bronxdale   
16           BRONX                                 Claremont-Bathgate   
17           BRONX                                         Highbridge   
18           BRONX                                 Woodlawn-Wakefield   
19           BRONX                                        Hunts Point   
20           BRONX                                  Crotona Park East   
21           BRONX                                Kingsbridge Heights   
22           BRONX                                           Longwood   
23           BRONX                                 Soundview-Bruckner   
24           BRONX    Soundview-Castle Hill-Clason Point-Harding Park   
25           BRONX                    Eastchester-Edenwald-Baychester   
26           BRONX                                        Parkchester   
27           BRONX                              Westchester-Unionport   
28           BRONX                                     Pelham Parkway   
29           BRONX            Van Nest-Morris Park-Westchester Square   
30           BRONX                                         Co-op City   
31           BRONX                         Spuyten Duyvil-Kingsbridge   
32           BRONX                Pelham Bay-Country Club-City Island   
33           BRONX           Schuylerville-Throgs Neck-Edgewater Park   
34           BRONX                            Allerton-Pelham Gardens   
35           BRONX                North Riverdale-Fieldston-Riverdale   
36           BRONX                            park-cemetery-etc-Bronx   
37           BRONX                                 Marble Hill-Inwood   
38        BROOKLYN                                Crown Heights North   
39        BROOKLYN                                      East New York   
40        BROOKLYN                                           Flatbush   
41        BROOKLYN                                        Brownsville   
42        BROOKLYN                  Prospect Lefferts Gardens-Wingate   
43        BROOKLYN                               Rugby-Remsen Village   
44        BROOKLYN                                 Stuyvesant Heights   
45        BROOKLYN                             East Flatbush-Farragut   
46        BROOKLYN                                         Ocean Hill   
47        BROOKLYN                                           Canarsie   
48        BROOKLYN                   East New York (Pennsylvania Ave)   
49        BROOKLYN                                     Bushwick South   
50        BROOKLYN                                            Bedford   
51        BROOKLYN                                            Erasmus   
52        BROOKLYN                                          Flatlands   
53        BROOKLYN                                Crown Heights South

In [31]:
#get the percentage of evictions for boroughs
top_NTA['%'] = top_NTA['Eviction_count'].div(top_NTA.groupby('Year')['Eviction_count'].transform('sum')).mul(100)
top_NTA.sort_values(by=['Year','%'], ascending=False)

Year        Borough                       Neighborhood_Tabulation_Area  \
874   2022       BROOKLYN                                Crown Heights North   
875   2022       BROOKLYN                                           Flatbush   
876   2022       BROOKLYN                  Prospect Lefferts Gardens-Wingate   
838   2022          BRONX                           Williamsbridge-Olinville   
877   2022       BROOKLYN                                        Brownsville   
878   2022       BROOKLYN                                      East New York   
879   2022       BROOKLYN                                     Bushwick South   
924   2022      MANHATTAN                                            Clinton   
880   2022       BROOKLYN                             East Flatbush-Farragut   
881   2022       BROOKLYN                               Rugby-Remsen Village   
839   2022          BRONX                         Bedford Park-Fordham North   
840   2022          BRONX                   East Concourse-Concourse Village   
925   2022      MANHATTAN                  Central Harlem North-Polo Grounds   
882   2022       BROOKLYN                                         Ocean Hill   
841   2022          BRONX                             Mott Haven-Port Morris   
883   2022       BROOKLYN                                           Canarsie   
926   2022      MANHATTAN                              Midtown-Midtown South   
927   2022      MANHATTAN                           Washington Heights South   
884   2022       BROOKLYN                                Crown Heights South   
842   2022          BRONX                                         Mount Hope   
885   2022       BROOKLYN                                            Erasmus   
1008  2022  STATEN ISLAND          West New Brighton-New Brighton-St. George   
843   2022          BRONX                                       East Tremont   
886   2022       BROOKLYN                                 Stuyvesant Heights   
844   2022          BRONX                                            Norwood   
845   2022          BRONX                                     West Concourse   
928   2022      MANHATTAN                                    Upper West Side   
846   2022          BRONX                              Van Cortlandt Village   
847   2022          BRONX                             West Farms-Bronx River   
887   2022       BROOKLYN                                          Bay Ridge   
929   2022      MANHATTAN                           Washington Heights North   
888   2022       BROOKLYN                                            Bedford   
930   2022      MANHATTAN         Hudson Yards-Chelsea-Flatiron-Union Square   
848   2022          BRONX                  University Heights-Morris Heights   
889   2022       BROOKLYN                   East New York (Pennsylvania Ave)   
951   2022         QUEENS               Hunters Point-Sunnyside-West Maspeth   
890   2022       BROOKLYN                                     Bushwick North   
931   2022      MANHATTAN                        Lenox Hill-Roosevelt Island   
849   2022          BRONX                                            Belmont   
932   2022      MANHATTAN                                   Hamilton Heights   
952   2022         QUEENS                                    Jackson Heights   
953   2022         QUEENS                                            Jamaica   
933   2022      MANHATTAN                  Battery Park City-Lower Manhattan   
934   2022      MANHATTAN                                  East Harlem North   
935   2022      MANHATTAN                                          Yorkville   
891   2022       BROOKLYN                                   Bensonhurst West   
892   2022       BROOKLYN                                          Flatlands   
893   2022       BROOKLYN                            Cypress Hills-City Line   
894   2022       BROOKLYN                                   Sunset Park East   
954   

In [32]:
#count the unique ntas in each borough
NTA_total_counts = no_2023.groupby(['Year','Borough']).Neighborhood_Tabulation_Area.nunique().to_frame('NTA_count')
NTA_total_counts

NTA_count
Year Borough                 
2017 BRONX                 38
     BROOKLYN              51
     MANHATTAN             29
     QUEENS                56
     STATEN ISLAND         18
2018 BRONX                 38
     BROOKLYN              50
     MANHATTAN             29
     QUEENS                57
     STATEN ISLAND         18
2019 BRONX                 38
     BROOKLYN              51
     MANHATTAN             28
     QUEENS                57
     STATEN ISLAND         18
2020 BRONX                 37
     BROOKLYN              51
     MANHATTAN             28
     QUEENS                53
     STATEN ISLAND         16
2021 BRONX                  9
     BROOKLYN              29
     MANHATTAN             14
     QUEENS                14
     STATEN ISLAND         11
2022 BRONX                 36
     BROOKLYN              50
     MANHATTAN             27
     QUEENS                57
     STATEN ISLAND         18

In [33]:
# turned my NTA_counts into a dataframe to use to find the percentages of the evictions counts for each year by borough
NTA_counts = no_2023.groupby(['Year','Borough']).Neighborhood_Tabulation_Area.nunique().reset_index(name='count')

NTA_counts['%'] = NTA_counts['count'].div(NTA_counts.groupby('Year')['count'].transform('sum')).mul(100)
NTA_counts.sort_values(by=['Year','%'], ascending=True)


Year        Borough  count          %
4   2017  STATEN ISLAND     18   9.375000
2   2017      MANHATTAN     29  15.104167
0   2017          BRONX     38  19.791667
1   2017       BROOKLYN     51  26.562500
3   2017         QUEENS     56  29.166667
9   2018  STATEN ISLAND     18   9.375000
7   2018      MANHATTAN     29  15.104167
5   2018          BRONX     38  19.791667
6   2018       BROOKLYN     50  26.041667
8   2018         QUEENS     57  29.687500
14  2019  STATEN ISLAND     18   9.375000
12  2019      MANHATTAN     28  14.583333
10  2019          BRONX     38  19.791667
11  2019       BROOKLYN     51  26.562500
13  2019         QUEENS     57  29.687500
19  2020  STATEN ISLAND     16   8.648649
17  2020      MANHATTAN     28  15.135135
15  2020          BRONX     37  20.000000
16  2020       BROOKLYN     51  27.567568
18  2020         QUEENS     53  28.648649
20  2021          BRONX      9  11.688312
24  2021  STATEN ISLAND     11  14.285714
22  2021      MANHATTAN     14  18.181818
23  2021         QUEENS     14  18.181818
21  2021       BROOKLYN     29  37.662338
29  2022  STATEN ISLAND     18   9.574468
27  2022      MANHATTAN     27  14.361702
25  2022          BRONX     36  19.148936
26  2022       BROOKLYN     50  26.595745
28  2022         QUEENS     57  30.319149

In [34]:
#**Conclusion**

#(Template)

#In 2017, there was anywhere between 18 to 56 Neighborhood Tabulation Areas affected by evicitons for each borough. The top two boroughs with about 30% of the eviction rate for the year were:
#* QUEENS with 56 effected Neighborhood Tabulation Areas, that's about 29.2% of the total evictions for the year.
#* BROOKLYN with 51 effected Neighborhood Tabulation Areas, that's about 26.6% of the total evictions for the year.


**Question** - Which part of the year are evictions high? <a class="anchor" id="15"></a>

In [38]:
#creating new column called Month
df_subset['Month'] = df_subset['Executed_Date'].dt.month_name()
# getting the eviction count for each month per year
Monthly_count = no_2023.groupby('Year').Month.value_counts().reset_index(name='Eviction count')
Monthly_count

Year      Month  Eviction count
0   2017    January            2064
1   2017        May            1953
2   2017     August            1937
3   2017    October            1917
4   2017      March            1893
5   2017       June            1876
6   2017      April            1750
7   2017   November            1683
8   2017  September            1668
9   2017       July            1612
10  2017   February            1589
11  2017   December            1266
12  2018    January            2141
13  2018        May            2011
14  2018     August            1862
15  2018      April            1835
16  2018      March            1819
17  2018       June            1778
18  2018       July            1730
19  2018   February            1694
20  2018    October            1670
21  2018  September            1529
22  2018   November            1410
23  2018   December            1130
24  2019    January            1957
25  2019        May            1691
26  2019      April            1651
27  2019   February            1590
28  2019      March            1570
29  2019       July            1562
30  2019       June            1501
31  2019     August            1442
32  2019    October            1349
33  2019  September            1320
34  2019   November            1063
35  2019   December             857
36  2020    January            1474
37  2020   February            1086
38  2020      March             566
39  2020   December              11
40  2020   November               6
41  2021       July              44
42  2021     August              38
43  2021   November              34
44  2021   December              26
45  2021    October               9
46  2021  September               7
47  2021    January               4
48  2021        May               4
49  2021      April               3
50  2021   February               3
51  2021       June               3
52  2021      March               1
53  2022   November             771
54  2022    October             644
55  2022  September             527
56  2022     August             478
57  2022   December             447
58  2022       June             394
59  2022       July             384
60  2022        May             383
61  2022      April             282
62  2022      March             282
63  2022   February             180
64  2022    January             107

**Conclusion** 

January is the top month that evictions occur in every year except 2021. Then May was the second top month that evictions were on the rise before the pandemic hit and the eviction moratorium went into effect in 2020.

## Overall Conclusion <a class="anchor" id="16"></a>

## Recommendation <a class="anchor" id="17"></a>

Based on my analysis, guiding questions and what what the client is interested in finding out, these are my recommendations.